Mo changelog mo problem (#47978)
CleverRaven/Cataclysm-DDA: data/changelog.txt

# 0.F (2020-12-01)

## Features:

> Added option to randomize INITIAL_DAY.

Reviewer: ```suggestion
Adds option to randomize INITIAL_DAY.
```

> Added ACTIVE enchantment condition.
> Allow recipes to specify individual activity levels.
> Add weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.

Reviewer: ```suggestion
Adds ACTIVE enchantment condition.
Allows recipes to specify individual activity levels.
Adds weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.
```

> Add weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.
> Report faction camp build sites and improve tutorial.

Reviewer: ```suggestion
Adds weariness, a sort of long-term stamina that limits characters' ability to be extremely active 24/7.
Report faction camp build sites and improve faction camp tutorial.
```
Am I correct re "faction camp" regarding the tutorial? (I've never built one.)

Author: I was actually puzzeled by that too, but now that you mention it, that's what it is.

> Added ACTIVE enchantment condition.
> Allow recipes to specify individual activity levels.
> Add weariness, a sort of long term stamina that limits the character's ability to be extremely active 24/7.
> Report faction camp build sites and improve tutorial.

Author: ```suggestion
Adds ACTIVE enchantment condition.
Allows recipes to specify individual activity levels.
Adds weariness, a sort of long term stamina that limits characters' ability to be extremely active 24/7.
Reports faction camp build sites and improves faction camp tutorial.
```

Feral Humans with blob psychosis.

> Allow plank and branch bundles to be made from cordage, vines, or wire.

Reviewer: BTW, regarding feral humans, I've been trying to figure out how best to have some of the more-sapient variety. Getting them to use guns appropriately looks to be a difficulty.
```suggestion
Allows plank and branch bundles to be made from cordage, vines, or wire.
```

Microlabs can contain artifacts in containment.
Adds NPC_class and NPC json templates and instructions.

> Adapts almost all vehicle definitions to use directional parts (most of the work by Curstwist).

Reviewer: ```suggestion
Adapts almost all vehicle definitions to use directional parts (most work done by Curstwist).
```

> Adapts almost all vehicle definitions to use directional parts (most of the work by Curstwist).

Author: ```suggestion
Adapts almost all vehicle definitions to use directional parts.
```

Update UltiCa tileset and make it default.
Adds a number of flying zombies.
Adds a new alternate needs panel to the label mode.
Added many proficiencies and uses for them.
Adds frog mother monster.
Adds mega mutant monsters.
Adds a selection of mutated woodland creatures.
New 'phase skulker' zombies teleport and haunt between the hallways of microlabs.
Make mutant organs and lungs contain toxins.

> Adds Streetlights to sidewalks; traffic lights, stop lines and pedestrian crossings to intersections.
> Adds Rust Zombies to the Game.
> Diversify venom sources and effects.

Reviewer: ```suggestion
Adds streetlights to sidewalks, and traffic lights, stop lines, and pedestrian crossings to intersections.
Adds Rust Zombies.
Diversifies venom sources and effects.
```

Reviewer 2: keep the semicolon otherwise the sentence structure's readability dips

Persist preferred aiming mode.
Makes Chibi Ultica a stand-alone tileset.
Brings back MSX+.
Add item length and pocket volume entries to menus.

> Add PAGE_UP/DOWN-fast-scroll to many menus with long lists.

Reviewer 3: ```suggestion
Adds fast scroll to many menus with long lists.
```

> NewCharacter - Add responsive screen behavior for small terminal sizes.

Reviewer 3: ```suggestion
Add responsive screen behavior to new character menu for small terminal sizes.
```

Reviewer: ```suggestion
Adds responsive screen behavior for small terminal sizes to new character menu.
```

Wielding combines item stacks if possible.
Robots can no longer be disabled through melee hits.
Fixes auto-consume zones in vehicles.

> Fix monster special attack range limiting code.

Reviewer: ```suggestion
Fixes monster special attack range-limiting code.
```

> Falling vehicles: prevent bouncing after the first collision with the ground.
> 
> ## Performance:
> Add option to enable batching when it is available in SDL.

Reviewer: ```suggestion
Falling vehicles: prevents bouncing after the first collision with the ground.

## Performance:
Adds an option to enable batching when it is available in SDL.
```

> Greatly reduce overhead of processing idle vehicles.

Reviewer: ```suggestion
Greatly reduces overhead from processing idle vehicles.
```

Optimizes Creature::has_effect (performance improved ~9.5%)
Adds a specialization for static/interned string_ids  (performance improved ~30%))

> Cache translations and _() macro (LOCALIZE=1 build perf improved ≈20%)

Reviewer 3: ```suggestion
Improve performance of builds with enabled localization by ≈20% by caching translations.
```

Reviewer: ```suggestion
Improves performance of builds with localization enabled by ~20% via caching translations.
```

> Major optimization in field processing.
> Add reachability cache/filter that can reject bresenham-based LOS checks in O(1) time.
> Use field_cache in map::get_field to speedup fire check in npc::assess_danger.

Reviewer: ```suggestion
Major optimization of field processing.
Adds reachability cache/filter that can reject bresenham-based LOS checks in O(1) time.
Uses field_cache in map::get_field to speedup fire check in npc::assess_danger.
```

> Speedup quicksave (mapbuffer::save) by limiting `ui_manager::redraw()` rate

Reviewer 3: ```suggestion
Speedup quicksave by limiting ui redraw rate
```

## Infrastructure:

> Add basic infrastructure for diseases.

Reviewer: ```suggestion
Adds basic infrastructure for diseases.
```
